### PR TITLE
misc: replace name with pname&version

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
#103997

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/emulators/fakenes/default.nix [hunk 1]
```diff
@@ -2,7 +2,9 @@
   libXxf86vm, libXcursor, libXpm }:
 
 stdenv.mkDerivation {
-  name = "fakenes-0.5.9b3";
+  pname = "fakenes";
+  version = "0.5.9b3";
+
   src = fetchurl {
     url = "mirror://sourceforge/fakenes/fakenes-0.5.9-beta3.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/fakenes/default.nix:9]:
```suggestion
  version = "0.5.9-beta3";

  src = fetchurl {
    url = "mirror://sourceforge/fakenes/fakenes-${version}.tar.gz";
```

File: pkgs/misc/screensavers/xss-lock/default.nix [hunk 2]
```diff
@@ -2,7 +2,8 @@
 , libXau, libXdmcp, xcbutil }:
 
 stdenv.mkDerivation {
-  name = "xss-lock-git-2018-05-31";
+  pname = "xss-lock-git";
+  version = "2018-05-31";
```
Comment (SuperSandro2000) [inline pkgs/misc/screensavers/xss-lock/default.nix:6]:
```suggestion
  pname = "xss-lock";
  version = "unstable-2018-05-31";
```

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 133674` at 65940a4b run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/d132e480c7e32e5e22300d8d5c864462">fakenes</a></li>
  </ul>
</details>
<details>
  <summary>17 packages built successfully:</summary>
  <ul>
    <li>bsod</li>
    <li>cups-bjnp</li>
    <li>dosbox</li>
    <li>foo2zjs</li>
    <li>foomatic-filters</li>
    <li>gutenprint</li>
    <li>m33-linux</li>
    <li>slock</li>
    <li>solfege</li>
    <li>splix</li>
    <li>uae</li>
    <li>vice</li>
    <li>vlock</li>
    <li>xlockmore</li>
    <li>xss-lock</li>
    <li>xtrlock-pam</li>
    <li>xwiimote</li>
  </ul>
</details>
<details>
  <summary>6 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/misc/drivers/m33-linux/default.nix:22:5:
```
   |
22 |     license = licenses.gpl2;
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/misc/solfege/default.nix:20:5:
```
   |
20 |     ./webbrowser.patch
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/misc/drivers/m33-linux/default.nix:14:3:
```
   |
14 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/misc/solfege/default.nix:18:5:
```
   |
18 |     ./menubar.patch
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/misc/solfege/default.nix:19:5:
```
   |
19 |     ./texinfo.patch
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/misc/solfege/default.nix:17:5:
```
   |
17 |     ./css.patch
   |     ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/misc/screensavers/xss-lock/default.nix [hunk 3]
```diff
@@ -2,7 +2,8 @@
 , libXau, libXdmcp, xcbutil }:
 
 stdenv.mkDerivation {
-  name = "xss-lock-git-2018-05-31";
+  pname = "xss-lock-git";
```
Comment (SuperSandro2000) [inline pkgs/misc/screensavers/xss-lock/default.nix:5]:
```suggestion
  pname = "xss-lock";
```

**Comment by Lassulus** [comment]
why not use `${pname}-${version}` in the urls where possible?

**Comment by Stunkymonkey** [comment]
i think a variable should be used, when the content changes often or is unpredictable. in our cases the package name does not really change, but the version does. Additionally by having a single variable; it might be easier to debug the URL if something is broken.

**Comment by Lassulus** [comment]
ok, sonds sensible

**Comment by Lassulus** [comment]
Result of `nixpkgs-review pr 133674` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>fakenes</li>
  </ul>
</details>
<details>
  <summary>23 packages built:</summary>
  <ul>
    <li>brgenml1cupswrapper</li>
    <li>brgenml1lpr</li>
    <li>bsod</li>
    <li>cups-bjnp</li>
    <li>dosbox</li>
    <li>epson-workforce-635-nx625-series</li>
    <li>foo2zjs</li>
    <li>foomatic-filters</li>
    <li>gutenprint</li>
    <li>keen4</li>
    <li>m33-linux</li>
    <li>slock</li>
    <li>solfege</li>
    <li>splix</li>
    <li>uae</li>
    <li>vice</li>
    <li>vlock</li>
    <li>wxmupen64plus</li>
    <li>xlockmore</li>
    <li>xss-lock</li>
    <li>xtrlock-pam</li>
    <li>xwiimote</li>
    <li>zsnes</li>
  </ul>
</details>

**Pull request merged**